app/controllers/note_controller.rb

```diff
@@ -0,0 +1,39 @@
+class NoteController < ApplicationController
+
+  def index
+    if current_user
+      @notes = current_user.notes
+    else
+      @notes = Note.all
+    end
+      render json: @notes
+  end
+
+  def create
+    @note = Note.new(note_params)
+    if @note.save
+      render json: @note
+    else
+      render json: {errors: @note.errors.full_messages.collect { |e| {error: e}}}, status: 400
+    end
+
+  end
+
+  def by_tag_name
+    @tags = Tag.find_by(name: params[:name])
+    render json: @tags, include: ['notes.tags', 'notes.user']
+  end
+
+  private
+
+  def note_params
+    pre_tags = params.permit(:title, :body, :tags, :user)
```

Comment:
Don't allow a user to be set directly ... it will allow any user to post as any other user (bonus: right now, if you pass a user param, you'll throw a 500)